src/Test/PdbUtilities/Reader/MockSymUnmanagedReader.cs

```diff
@@ -321,7 +321,7 @@ public MockSymUnmanagedNamespace(string name)
             if (name != null)
             {
                 var builder = ArrayBuilder<char>.GetInstance();
```

Comment:
This whole block seems more complicated than it needs to be.

```suggestion
                var builder = ArrayBuilder<char>.GetInstance(name.Length + 1);
```

However, if `fixed` in C# produces a `\0`-terminated string, you can just do this:

```
fixed (char* namePtr = name)
{
  _nameChars = ImmutableArray.Create(new ReadOnlySpan<char>(namePtr, name.Length + 1));
}
```